app/controllers/comments_controller.rb

```diff
@@ -87,18 +87,26 @@ def check_if_restricted
     redirect_to new_user_session_path(restricted_commenting: true)
   end
 
-  # Check to see if the ultimate_parent is a Work, and if so, if it allows
+  # Check to see if the ultimate_parent is a Work or AdminPost, and if so, if it allows
   # comments for the current user.
   def check_parent_comment_permissions
     parent = find_parent
-    return unless parent.is_a?(Work)
+    if parent.is_a?(Work)
+      entity = "work"
+      path = proc { |x| work_path(x) }
```

Comment:
I was wondering why I haven't seen code like this in the repo, turned out that we use `send` a lot, e.g.

https://github.com/otwcode/otwarchive/blob/b5bf2e7a5cd362d1e350f2be087e8049721a2484/app/controllers/users_controller.rb#L267-L271

Do you know if calling `proc`s is faster than sending?

Reply:
So as a ruby newbie, I was looking for a way to assign a function to a variable, and I didn't know about `proc` vs `send`.

That said, looking back on this, I don't think I need any of those, since we're always sending back to the parent anyway.